nav2_behavior_tree/CMakeLists.txt

```diff
@@ -229,39 +240,59 @@ set(GENERATED_DIR ${CMAKE_CURRENT_BINARY_DIR}/gen)
 configure_file(plugins_list.hpp.in ${GENERATED_DIR}/plugins_list.hpp)
 
 add_executable(generate_nav2_tree_nodes_xml src/generate_nav2_tree_nodes_xml.cpp)
-ament_target_dependencies(generate_nav2_tree_nodes_xml ${dependencies})
+target_link_libraries(generate_nav2_tree_nodes_xml PRIVATE
+  behaviortree_cpp::behaviortree_cpp
+)
 # allow generate_nav2_tree_nodes_xml to find plugins_list.hpp
 target_include_directories(generate_nav2_tree_nodes_xml PRIVATE ${GENERATED_DIR})
 install(TARGETS generate_nav2_tree_nodes_xml DESTINATION lib/${PROJECT_NAME})
 
-
-install(DIRECTORY include/
-  DESTINATION include/
+install(TARGETS ${plugin_libs}
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
 )
 
-install(DIRECTORY test/utils/
-        DESTINATION include/${PROJECT_NAME}/utils/
+install(DIRECTORY include/
+  DESTINATION include/${PROJECT_NAME}
 )
 
 install(FILES nav2_tree_nodes.xml DESTINATION share/${PROJECT_NAME})
-install(FILES ${GENERATED_DIR}/plugins_list.hpp DESTINATION include/${PROJECT_NAME})
+install(FILES ${GENERATED_DIR}/plugins_list.hpp DESTINATION include/${PROJECT_NAME}/${PROJECT_NAME})
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
   find_package(ament_cmake_gtest REQUIRED)
+
+  ament_find_gtest()
+
   add_subdirectory(test)
 endif()
 
 ament_export_include_directories(
-  include
+  include/${PROJECT_NAME}
 )
 
-ament_export_libraries(
-  ${library_name}
-  ${plugin_libs}
-)
+ament_export_libraries(${library_name})
```

Comment:
Shouldn't we also export the plugins? Someone could plausibly use existing plugins as a base class and override methods for custom implementations. It would surprise me if _someone_ wasn't already doing that.

Reply:
Ah, OK.  I wasn't sure about that, so I did not export them.  But I absolutely can, I'll look into doing that.

Reply:
This is fixed in the latest push.